src/scripts/party/PartyPokemon.ts

```diff
@@ -283,6 +283,34 @@ class PartyPokemon implements Saveable {
         GameHelper.incrementObservable(player.itemList[vitaminName], amount);
     }
 
+    public useConsumable(type: GameConstants.ConsumableType, amount: number): void {
+        const itemName = GameConstants.ConsumableType[type];
+        if (!player.itemList[itemName]()) {
+            return Notifier.notify({
+                message : `You do not have any more ${ItemList[itemName].displayName}`,
+                type : NotificationConstants.NotificationOption.danger,
+            });
+        }
+        switch (type) {
+            case GameConstants.ConsumableType.RareCandy : amount = Math.min(amount, player.itemList[itemName]());
+                const curAttack = this.calculateAttack(true);
+                GameHelper.incrementObservable(this._attackBonusPercent, 25 * amount);
```

Comment:
Can we move this code to the rare candy code? Make an overrideable method in consumables called `use` or something.
Will make it easier to see what rare candy does

Reply:
I am not totally sure if this is a good idea or not... It sounds good in theory, but it might look messy, if we move it...

Reply:
You know what ? I did it that way first lol. But I did not like it and thought you would not neither. I passed the function through the constructor. Although it needed to import much more things which had nothing to do with models. I thought it would be messy for no reason and ended up putting that `useConsumable` in whom we can stack the processes in a proper way.